packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -155,13 +181,22 @@ def _serial_add_to_archive(
 
 
 async def archive_dir(
-    dir_to_compress: Path, destination: Path, compress: bool, store_relative_path: bool
+    dir_to_compress: Path,
+    destination: Path,
+    compress: bool,
+    store_relative_path: bool,
+    exclude_paths: List[Path] = None,
 ) -> None:
     """
     When archiving, undecodable bytes in filenames will be escaped,
     zipfile does not like them.
     When unarchiveing, the **escaped version** of the file names
     will be created.
+
+    The **exclude_paths** is a list of paths of absolute or relative
+    files which will be match respectively the begining or the end
+    of a file. If such a match exists that file will be excluded
+    form the archive.
```

Comment:
maybe you could follow how a ```.dockerignore``` or ```.gitignore``` file works. it might bring more lights on how this is supposed to work. Then these would be ```exclude_patterns``` and it might be easier to understand.

Reply:
I agree with @sanderegg . 
file path filtering (either include/exclude) in inherent to the ``pathlib`` library that uses [glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) and even provides a function to iterate them https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

Check the tip above on ``fnmatch`` module. Read carefully all its functions (it has just a few but it is nicely and simply designed)